cordova-lib/src/plugman/fetch.js

```diff
@@ -146,11 +145,7 @@ function fetchPlugin(plugin_src, plugins_dir, options) {
         });
     }).then(function(result){
         checkID(options.expected_id, result.pinfo);
-        var data = { source: result.fetchJsonSource };
-        data.is_top_level = options.is_top_level; 
-        data.variables = options.variables || {};
-        metadata.save_fetch_metadata(plugins_dir, result.pinfo.id, data);
-        return result.dest;
+        return result;
```

Comment:
Function name `checkID` seems to do something and return a result, but you don't capture that result, nor do you return anything with that result.  Because you pass the value of `result.pinfo` by value, `checkID` has no opportunity to mutate or modify anything.

What happens if `checkID` "fails", if it can?  If it can't, why is it called `check`?